modules/beta-autopilot-private-cluster/cluster.tf

```diff
@@ -0,0 +1,219 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file was automatically generated from a template in ./autogen/main
+
+/******************************************
+  Create Container Cluster
+ *****************************************/
+resource "google_container_cluster" "primary" {
+  provider = google-beta
+
+  name            = var.name
+  description     = var.description
+  project         = var.project_id
+  resource_labels = var.cluster_resource_labels
+
+  location          = local.location
+  node_locations    = local.node_locations
+  cluster_ipv4_cidr = var.cluster_ipv4_cidr
+  network           = "projects/${local.network_project_id}/global/networks/${var.network}"
+
+  dynamic "release_channel" {
+    for_each = local.release_channel
+
+    content {
+      channel = release_channel.value.channel
+    }
+  }
+
+  subnetwork = "projects/${local.network_project_id}/regions/${local.region}/subnetworks/${var.subnetwork}"
+
+  default_snat_status {
+    disabled = var.disable_default_snat
+  }
+  min_master_version = var.release_channel != null ? null : local.master_version
+
+  dynamic "cluster_telemetry" {
+    for_each = local.cluster_telemetry_type_is_set ? [1] : []
+    content {
+      type = var.cluster_telemetry_type
+    }
+  }
+  logging_service    = local.cluster_telemetry_type_is_set ? null : var.logging_service
+  monitoring_service = local.cluster_telemetry_type_is_set ? null : var.monitoring_service
+  vertical_pod_autoscaling {
+    enabled = var.enable_vertical_pod_autoscaling
+  }
+  enable_autopilot = var.enable_autopilot
+  dynamic "master_authorized_networks_config" {
+    for_each = local.master_authorized_networks_config
+    content {
+      dynamic "cidr_blocks" {
+        for_each = master_authorized_networks_config.value.cidr_blocks
+        content {
+          cidr_block   = lookup(cidr_blocks.value, "cidr_block", "")
+          display_name = lookup(cidr_blocks.value, "display_name", "")
+        }
+      }
+    }
+  }
+
+  master_auth {
+    username = var.basic_auth_username
```

Comment:
I dont think it is possible to use basic auth and will anyway be removed 1.19+, we could drop this.
@morgante any thoughts?

Reply:
separate PR/issue?